pkg/controller/backup_controller_test.go

```diff
@@ -422,3 +426,150 @@ func TestProcessBackup(t *testing.T) {
+
+	tests := []struct {
+		name             string
+		backup           *arktest.TestBackup
+		locations        []*arktest.TestVolumeSnapshotLocation
+		expectedSuccess  bool
+		defaultLocations map[string]string
+	}{
+		{
+			name:            "location name does not correspond to any existing location",
+			backup:          arktest.NewTestBackup().WithName("backup1").WithPhase(v1.BackupPhaseNew).WithVolumeSnapshotLocations([]string{"random-name"}),
+			locations:       arktest.NewTestVolumeSnapshotLocation().WithName(dupLocationNames[0]).WithProviderConfig(dupLocationList),
+			expectedSuccess: false,
+		},
+		{
+			name:            "duplicate locationName per provider: should filter out dups",
+			backup:          arktest.NewTestBackup().WithName("backup1").WithPhase(v1.BackupPhaseNew).WithVolumeSnapshotLocations(dupLocationNames),
+			locations:       arktest.NewTestVolumeSnapshotLocation().WithName(dupLocationNames[0]).WithProviderConfig(dupLocationList),
+			expectedSuccess: true,
+		},
+		{
+			name:            "multiple location names per provider",
+			backup:          arktest.NewTestBackup().WithName("backup1").WithPhase(v1.BackupPhaseNew).WithVolumeSnapshotLocations(multipleLocationNames),
+			locations:       arktest.NewTestVolumeSnapshotLocation().WithName(multipleLocationNames[0]).WithProviderConfig(multipleLocationList),
+			expectedSuccess: false,
+		},
+		{
+			name:             "no location name for the provider exists: the provider's default should be added",
+			backup:           arktest.NewTestBackup().WithName("backup1").WithPhase(v1.BackupPhaseNew),
+			defaultLocations: defaultLocationAWS,
+			expectedSuccess:  true,
+		},
+		{
+			name:            "no existing location name and no default location name given",
+			backup:          arktest.NewTestBackup().WithName("backup1").WithPhase(v1.BackupPhaseNew),
+			expectedSuccess: true,
+		},
+		{
+			name:             "multiple location names for a provider, default location name for another provider",
+			backup:           arktest.NewTestBackup().WithName("backup1").WithPhase(v1.BackupPhaseNew).WithVolumeSnapshotLocations(dupLocationNames),
+			locations:        arktest.NewTestVolumeSnapshotLocation().WithName(dupLocationNames[0]).WithProviderConfig(dupLocationList),
+			defaultLocations: defaultLocationGCP,
+			expectedSuccess:  true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			var (
+				client          = fake.NewSimpleClientset()
+				backupper       = &fakeBackupper{}
+				sharedInformers = informers.NewSharedInformerFactory(client, 0)
+				logger          = logging.DefaultLogger(logrus.DebugLevel)
+				pluginManager   = &pluginmocks.Manager{}
+				backupStore     = &persistencemocks.BackupStore{}
+			)
+			defer backupper.AssertExpectations(t)
+			defer pluginManager.AssertExpectations(t)
+			defer backupStore.AssertExpectations(t)
+
+			c := NewBackupController(
```

Comment:
it looks like the only field of `*backupController` that `defaultAndValidateSnapshotLocations` uses is the `snapshotLocationLister` - so I'd switch this to just say: 
```
c := &backupController{
  snapshotLocationLister: sharedInformers.Ark().V1().VolumeSnapshotLocations().Lister(),
}
```
and then get rid of all the extraneous dependencies that aren't actually used